framework/platform/qemu/run.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env nix-shell
+#!nix-shell --pure -p qemu -i bash
+
+# Check if a version argument is provided
+if [ -z "$1" ]; then
+    echo "Usage: $0 <qemu-version>"
+    exit 1
+fi
+
+qemu_version="$1"
+# -s -S -nographic\
+qemu-system-"$qemu_version" -nographic\
+    -M virt,secure=on,virtualization=on,gic-version=3 \
+   -cpu cortex-a53 -smp 4 -m 4G\
+   -bios ./result-2/bin/qemu-"$qemu_version"-virt/flash.bin \
+   -device loader,file="./result-3/bin/bao.bin",addr=0x50000000,force-raw=on\
```

Comment:
We need to make this more generic: `file="./result-3/bin/bao.bin"`. Since the `result-3 `folder isn't static and can change, it would be better to search for the `bao.bin` file in the compilation folders and then pass the path where it was found to QEMU. This approach would make it agnostic to the specific name of the `results-x` folders.

Reply:
Fully agree, that's a really good point. Adressed in commit . Thanks!
The same goes for the ``-bios``, so I took the liberty to incorporate the same approach.